lib/gollum-lib/markups.rb

```diff
@@ -6,7 +6,7 @@ class Markup
         Kramdown::Document.new(content, :auto_ids => false, :input => "markdown").to_html
     }
 
-    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/)
+    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/, :h1 => /^(#\s+)(\w+.*)/)
```

Comment:
Ah, that makes clear what you're after. I think something like `Page#edited_page` doesn't really fit gollum, since it doesn't play well with git -- it assumes we can edit the `Page`'s content, then save it to the repo, but there really isn't anything like a persisting, unsaved-but-modified page due to git's architecture (different content = different blob). That makes for less than ideally OO code, but that's the price of having git as a backend. :)

So since a `Page`'s content is basically static, I think there needs to be a method that returns the a page's content, split into `before`, `wanted section`, and `after`. That array can then be used in the frontend to display only the `wanted section`, joining it again with `before` and `after` when the form is submitted.

Nevertheless, interesting setup, and this:
```ruby
    @start_offset = extract_token(real_sections[section_number - 1])
    @end_offset = extract_token(real_sections[section_number])
```
Achieves what I was doing in a much cleaner way. :+1: I'll play around a bit and try to integrate as much of @SkyCrawl's and @bartkamphorst's suggestions.